giraffe/src/components/geo/PointMapLayer.tsx

```diff
@@ -0,0 +1,201 @@
+  for (const marker of cluster._markers) {
+    result.push(marker.value)
+    clusterRenderingProperties = marker.clusterRenderingProperties
+  }
+  for (const childCluster of cluster._childClusters) {
+    const properties = getClusterValues(childCluster, result)
+    if (properties) clusterRenderingProperties = properties
+  }
+  return clusterRenderingProperties
+}
+
+const getClusterValue = (
+  cluster
+): {value: number; properties: GeoPointMapViewLayer} => {
+  const values: number[] = []
+  const properties = getClusterValues(cluster, values)
+  const aggregation =
+    properties.clusterAggregationFunction || ClusterAggregation.max
+  switch (aggregation) {
+    case ClusterAggregation.max:
+      return {value: Math.max(...values), properties}
+    case ClusterAggregation.min:
+      return {value: Math.min(...values), properties}
+    case ClusterAggregation.mean:
+      return {
+        value: values.reduce((a, x) => a + x, 0) / values.length,
+        properties,
+      }
+    case ClusterAggregation.median:
+      if (values.length === 1) return {value: values[0], properties}
+      values.sort()
+      if (values.length % 2 === 0) {
+        const half = values.length / 2
+        return {value: (values[half - 1] + values[half]) / 2, properties}
+      } else {
+        return {value: values[Math.floor(values.length / 2)], properties}
+      }
+    default:
+      return {value: 0, properties}
+  }
+}
+
+const createClusterCustomIcon = function(cluster) {
+  let outerColor = DEFAULT_CLUSTER_COLOR,
+    innerColor = 'white'
+  const {value, properties} = getClusterValue(cluster)
+  if (properties.areClustersColored && properties.colors) {
+    innerColor = 'white'
+    outerColor = getColor(properties.colors, value, false)
+  }
+  const outer = `\
+background-color: ${outerColor};\
+height: ${CLUSTER_ICON_SIZE}px;\
+border-radius: ${CLUSTER_ICON_SIZE / 2}px;`
+  const inner = `\
+height: ${CLUSTER_ICON_SIZE - 2 * CLUSTER_ICON_SIZE_BORDER}px;\
+width: ${CLUSTER_ICON_SIZE - 2 * CLUSTER_ICON_SIZE_BORDER}px;\
+left: ${CLUSTER_ICON_SIZE_BORDER}px;\
+top: ${CLUSTER_ICON_SIZE_BORDER}px;\
+position: relative;\
+border-radius:${(CLUSTER_ICON_SIZE - 2 * CLUSTER_ICON_SIZE_BORDER) / 2}px;\
+background-color: ${innerColor};\
+font-size: 13px;\
+text-align: center;\
+color: dimgray;\
+line-height:${CLUSTER_ICON_SIZE - 2 * CLUSTER_ICON_SIZE_BORDER}px;`
+  return L.divIcon({
+    html: `<div style="${outer}"><div style="${inner}">${cluster.getChildCount()}</div></div>`,
+    className: 'marker-cluster-custom',
+    iconSize: L.point(CLUSTER_ICON_SIZE, CLUSTER_ICON_SIZE, true),
+  })
+}
+
+export const PointMapLayer: FunctionComponent<Props> = props => {
+  const {table, colorFieldName, properties, stylingConfig, isClustered} = props
+  const rowCount = table.getRowCount()
+  const result = [],
+    tooltips = []
+  for (let i = 0; i < rowCount; i++) {
+    const latLon = table.getLatLon(i)
+    if (!latLon) continue
```

Comment:
Please always use braces for if statements. We will be updating the eslint rules to reflect this.